packages/grid/x-grid-data-generator/src/useDemoData.ts

```diff
@@ -85,7 +85,7 @@ export const useDemoData = (options: DemoDataOptions): DemoDataReturnType => {
   const [loading, setLoading] = React.useState(true);
 
   React.useEffect(() => {
-    const cacheKey = `${options.dataSet}-${rowLength}-${index}-${options.maxColumns}`;
+    const cacheKey = `${options.dataSet}-${options.rowLength}-${index}-${options.maxColumns}`;
```

Comment:
why options.rowLength?

Reply:
What alternatives do you have in mind?

Reply:
Well above we have
```
  const [rowLength, setRowLength] = React.useState(options.rowLength);
```
Not sure why you want to refactor as you can destructure `setRowLength` and achieve what you want.
A bit like a useState `[rowLength, setRowLength]`
This change involves refactoring all demos that uses `setRowLength` and move the state from this hook to the demos, so I don't think that bring any value

Reply:
Ok, this seems to be a topic in itself. I'll revert the change I made and keep it as it was. I'll open a separate PR dedicated just to that problem so that we don't discuss 2 separate topics here if you all agree

Reply:
Agree it is separate but I don't see it as an issue nor a priority.
What are you trying to solve? How about we open an issue?

Reply:
👍 for taking care of it in a different PR. The problem is described in https://github.com/mui-org/material-ui-x/pull/1199#discussion_r591361936. To make it short, there isn't a single source of truth, this forces us to keep two-state in sync.